services/httpd/handler.go

```diff
@@ -2564,6 +2593,33 @@ func parseCryptoDiagnostics(d *diagnostics.Diagnostics) (map[string]interface{},
 	return m, nil
 }
 
+// parseConfigDiagnostics converts the config diagnostics into proper format
+// for JSON marshaling, handling toml.Size and toml.Duration types properly.
+func parseConfigDiagnostics(d *diagnostics.Diagnostics) (map[string]interface{}, error) {
+	if len(d.Rows) == 0 {
+		return nil, fmt.Errorf("no config diagnostic data available")
+	}
+
+	m := make(map[string]interface{})
+	for i, col := range d.Columns {
+		if i >= len(d.Rows[0]) {
+			continue
+		}
+
+		val := d.Rows[0][i]
+		switch v := val.(type) {
+		case toml.Size:
+			m[col] = uint64(v)
+		case toml.Duration:
+			m[col] = time.Duration(v).String()
+		default:
+			m[col] = v
```

Comment:
There are no other interesting types that need conversion?

Reply:
The only other two that I saw were `toml.FileMode` and `toml.Group` which were just a uint64 each. I didn't see them used in the configuration options that I'm exposing either.